webapp-django/crashstats/manage/views.py

```diff
@@ -789,3 +792,140 @@ def _get_edit_url(action, extra):
         'batch_size': batch_size,
         'page': page,
     }
+
+
+@superuser_required
+def api_tokens(request):
+    all_possible_permissions = (
+        Permission.objects.filter(content_type__model='')
+        .order_by('name')
+    )
+    possible_permissions = []
+    for permission in all_possible_permissions:
+        possible_permissions.append(permission)
+
+    expires_choices = (
+        (1, '1 day'),
+        (7, '1 week'),
+        (30, '1 month'),
+        (30 * 3, '3 months'),
+        (365, '1 year'),
+        (365 * 10, '10 years'),
+    )
+
+    if request.method == 'POST':
+        form = forms.APITokenForm(
+            request.POST,
+            possible_permissions=possible_permissions,
+            expires_choices=expires_choices,
+        )
+        if form.is_valid():
+            data = form.cleaned_data
+            token = Token.objects.create(
+                user=data['user'],
+                notes=data['notes'],
+                expires=data['expires'],
+            )
+            for permission in data['permissions']:
+                token.permissions.add(permission)
+
+            messages.success(
+                request,
+                'API Token for %s created. ' % token.user.email
+            )
+            return redirect('manage:api_tokens')
```

Comment:
I don't understand your answer, here's the structure I see here: 

```python
if request.method == 'POST':
    if form.is_valid():
        # do something if the form is valid
else:
    # do something if not using POST
```

If ``form.is_valid()`` returns a falsy value, that's not handled. Unless I am not looking at it correctly, but then I'd love an explanation because it is not obvious. :)